Automattic/wp-calypso
# Plugins: Refactor PluginNotices to a React.Component

Currently, `PluginNotices` is declared as a plain object, and it's consumed as a mixin in `PluginsList` and `SinglePlugin`. Not only mixins (and their deprecated friend `createReactClass`) are deprecated; but this prevents from connecting the notices component properly to Redux and relying on the React component lifecycle in a more generic, composable way.

This PR refactors `PluginNotices` to a `React.Component`, updates its usage accordingly, and moves the file to `client/my-sites/plugins` as a sibling to the rest of the plugin components.

This paves the way for several upcoming refactors:
* Fully reduxifying `PluginNotices` by untangling `PluginsLog` from it.
* Get rid of `createReactClass` in plugins completely 🥳 

This PR is part of #24180.

#### Changes proposed in this Pull Request

* Plugins: Refactor `PluginNotices` to a `React.Component`
* Plugins: Use `PluginNotices` as a component in `SinglePlugin`
* Plugins: Use `PluginNotices` as a component in `PluginsList`
* Plugins: Move notices component to plugins dir
* Plugins: Fix and clean up `PluginsList` tests

#### Testing instructions

* Get a Jetpack site with some plugins for testing.
* Compare against production and verify that global notices in plugin items still work the same way when updating, installing, or removing plugins, as well as toggling auto-updates and activating/deactivating plugins:
  * `/plugins/:plugin/:site`
  * `/plugins/:plugin`
  * `/plugins`
  * `/plugins/manage`
  * `/plugins/manage/:site` - for a single plugin
  * `/plugins/manage/:site` - for multiple selected plugins (bulk actions are enabled by clicking the "Edit All" button)
* Verify all tests pass.
